src/Data/Tree/Binary/Zipper/Properties.agda

```diff
@@ -0,0 +1,145 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Tree Zipper-related properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Tree.Binary.Zipper.Properties where
+
+open import Data.Tree.Binary as BT using (Tree; node; leaf)
+open import Data.Tree.Binary.Zipper
+open import Data.Nat using (ℕ; suc; _+_)
+open import Data.Nat.Properties
+open import Data.List.Base as List using (List ; [] ; _∷_; sum)
+open import Data.Maybe.Base using (Maybe; just; nothing)
+open import Data.Maybe.Relation.Unary.All using (All; just; nothing)
+open import Relation.Binary.PropositionalEquality
+open ≡-Reasoning
+open import Function
+open import Level using (Level)
+
+private
+  variable
+    a b c d : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    D : Set d
+
+-- Invariant: Zipper represents a given tree
+
+-- Stability under moving
+
+toTree-up-identity : (zp : Zipper A B) → All ((_≡_ on toTree) zp) (up zp)
+toTree-up-identity (mkZipper [] foc) = nothing
+toTree-up-identity (mkZipper (leftBranch m l ∷ ctx) foc) = just refl
+toTree-up-identity (mkZipper (rightBranch m r ∷ ctx) foc) = just refl
+
+toTree-left-identity : (zp : Zipper A B) → All ((_≡_ on toTree) zp) (left zp)
+toTree-left-identity (mkZipper ctx (leaf x)) = nothing
+toTree-left-identity (mkZipper ctx (node l m r)) = just refl
+
+toTree-right-identity : (zp : Zipper A B) → All ((_≡_ on toTree) zp) (right zp)
+toTree-right-identity (mkZipper ctx (leaf x)) = nothing
+toTree-right-identity (mkZipper ctx (node l m r)) = just refl
+
+-- Tree-like operations indeed correspond to their counterparts
+------------------------------------------------------------------------
+
+toTree-#nodes-commute : ∀ (zp : Zipper A B) → #nodes zp ≡ BT.#nodes (toTree zp)
+toTree-#nodes-commute (mkZipper c v) = helper c v
+  where
+    helper : (cs : List (Crumb A B))
+           → (t : Tree A B)
+           → #nodes (mkZipper cs t) ≡ BT.#nodes (toTree (mkZipper cs t))
```

Comment:
The [style guide](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md) suggests that function arrows go at the end of lines rather than the beginning.